packages/web/components/templates/Menu.tsx

```diff
@@ -0,0 +1,171 @@
+import Link from 'next/link'
+import React, { useEffect, useState } from 'react'
+
+import { useGetLabelsQuery } from '../../lib/networking/queries/useGetLabelsQuery'
+import { useGetSubscriptionsQuery } from '../../lib/networking/queries/useGetSubscriptionsQuery'
+
+import { ProSidebar, Menu, MenuItem, SubMenu } from 'react-pro-sidebar'
+import { Tag } from 'phosphor-react'
+
+// styles
+const proSideBarStyles = {
+  display: 'inline-block',
+}
+
+// Types
+type MenuItems = {
+  label: string
+  query: string
+  active?: boolean
+  icon: string | JSX.Element | null
+  href: string
+}
+
+// Functions
+const calculateTodayMenuItem = () => {
+  const timeZoneHourDiff = -new Date().getTimezoneOffset() / 60
+  const hrefStr = `?q=in%3Ainbox+saved%3A${
+    new Date(new Date().getTime() - 24 * 3600000).toISOString().split('T')[0]
+  }Z%${timeZoneHourDiff}B2..*`
+  return hrefStr
+}
+
+const createDynamicMenuItems = (
+  labels: Array<any>,
+  subscriptions: Array<any>
+) => {
+  const labelsList: Array<MenuItems> = []
+  const subscriptionsList: Array<MenuItems> = []
+  // Create labels list
+  if (labels.length) {
+    labels.map((l) =>
+      labelsList.push({
+        label: l.name,
+        query: `label:"${l.name}"`,
+        icon: <Tag size={18} weight="light" />,
+        href: `?q=label:"${l.name}"`,
+      })
+    )
+  }
+  // create subscriptions list
+  if (subscriptions.length) {
+    subscriptions.map((s) =>
+      subscriptionsList.push({
+        label: s.name,
+        query: `subscription:"${s.subscription}"`,
+        icon: 'subscription',
+        href: `?q=subscription:"${s.subscription}"`,
+      })
+    )
+  }
+  return [{ labels: [...labelsList], subscriptions: [...subscriptionsList] }]
```

Comment:
I don't think this needs to be an array anymore. I think if we just return`{ labels: [...labelsList], subscriptions: [...subscriptionsList] }` some of the below code gets simpler.

Reply:
I just pushed a small change to do this.